### ISSUE TITLE
A question about spurious shifts experiments

### ISSUE THREAD

**Issue by mmdzb** [vita-epfl/causalmotion#1]
Thanks for sharing the code!

 I'm interested in this project and I'm trying to run the experiments based on the code and paper. 

However, in the spurious shifts experments, there is an interesting phenomenon.  In the experiments of λ (figure 4 in the paper), my result is opposite from that in the paper, which means that λ=1 reaches the lowest ADE and λ=5 reaches the highest ADE.

I think I have followed all the settings in the paper and code (e.g. version of pytorch, hyperparameters), so I'm a little confused about this phenomenon. Could you please give me some advice about this? Is there anything special that you think should be focused on when runing the spurious shifts experiments?

Sorry for bothering you and looking forward to replys.

**Comment by riccardocadei** [comment]
Hi @mmdzb, thank you for your interest and reporting the issue.

I have just rerun the experiment using `seed=1` as we left by default in the script [spurious/scripts/add_confidence/train.sh](https://github.com/vita-epfl/causalmotion/blob/main/spurious/scripts/add_confidence/train.sh). 

Here's our obtained result ![ade](https://user-images.githubusercontent.com/47679062/159342724-acd9c53a-e834-42c3-b993-861ce3c5d0a0.png)

The global pattern looks pretty similar to our previous figure, while there are still some local variations. We conjecture that the local variations are attributed to some sources of randomness, e.g., random sampling for prediction. However, running experiments over multiple seeds should reduce the variations, especially when considering a broader spectrum of λ.

Hope this helps. If you have any further questions, please don't hesitate to let me know.

**Comment by mmdzb** [comment]
Thanks for your reply! I'm trying to 

> Hi @mmdzb, thank you for your interest and reporting the issue.
> 
> I have just rerun the experiment using `seed=1` as we left by default in the script [spurious/scripts/add_confidence/train.sh](https://github.com/vita-epfl/causalmotion/blob/main/spurious/scripts/add_confidence/train.sh).
> 
> Here's our obtained result ![ade](https://user-images.githubusercontent.com/47679062/159342724-acd9c53a-e834-42c3-b993-861ce3c5d0a0.png)
> 
> The global pattern looks pretty similar to our previous figure, while there are still some local variations. We conjecture that the local variations are attributed to some sources of randomness, e.g., random sampling for prediction. However, running experiments over multiple seeds should reduce the variations, especially when considering a broader spectrum of λ.
> 
> Hope this helps. If you have any further questions, please don't hesitate to let me know.

**Comment by mmdzb** [comment]
Thanks for your reply! I will try to figure the problem out.
